pkg/manager/gomod.go

```diff
@@ -0,0 +1,338 @@
+package manager
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"unicode"
+
+	zglob "github.com/mattn/go-zglob"
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/protodep/pkg/modutils"
+	"github.com/solo-io/protodep/protodep"
+	"github.com/spf13/afero"
+)
+
+const (
+	ProtoMatchPattern   = "**/*.proto"
+	SoloKitMatchPattern = "**/solo-kit.json"
+)
+
+var (
+	// offer sane defaults for proto vendoring
+	DefaultMatchPatterns = []string{ProtoMatchPattern, SoloKitMatchPattern}
+
+	// matches ext.proto for solo hash gen
+	ExtProtoMatcher = &protodep.GoModImport{
+		Package:  "github.com/solo-io/protoc-gen-ext",
+		Patterns: []string{"extproto/*.proto"},
+	}
+
+	// matches validate.proto which is needed by envoy protos
+	EnvoyValidateProtoMatcher = &protodep.GoModImport{
+		Package:  "github.com/envoyproxy/protoc-gen-validate",
+		Patterns: []string{"validate/*.proto"},
+	}
+
+	// matches all solo-kit protos, useful for any projects using solo-kit
+	SoloKitProtoMatcher = &protodep.GoModImport{
+		Package:  "github.com/solo-io/solo-kit",
+		Patterns: []string{"api/**/*.proto", "api/" + SoloKitMatchPattern},
+	}
+
+	// matches gogo.proto, used for gogoproto code gen.
+	GogoProtoMatcher = &protodep.GoModImport{
+		Package:  "github.com/gogo/protobuf",
+		Patterns: []string{"gogoproto/*.proto"},
+	}
+
+	// default match options which should be used when creating a solo-kit project
+	DefaultMatchOptions = []*protodep.Import{
+		{
+			ImportType: &protodep.Import_GoMod{
+				GoMod: ExtProtoMatcher,
+			},
+		},
+		{
+			ImportType: &protodep.Import_GoMod{
+				GoMod: EnvoyValidateProtoMatcher,
+			},
+		},
+		{
+			ImportType: &protodep.Import_GoMod{
+				GoMod: SoloKitProtoMatcher,
+			},
+		},
+		{
+			ImportType: &protodep.Import_GoMod{
+				GoMod: GogoProtoMatcher,
+			},
+		},
+	}
+)
```

Comment:
what is the reasoning behind hard-coding dependencies on the projects that will import this? these are circular dependencies (and seems like the wrong place to define them)

Reply:
Leftover from solo-kit work, will remove